hbase-server/src/main/java/org/apache/hadoop/hbase/util/TableDescriptorChecker.java

```diff
@@ -210,6 +215,37 @@ private static void checkReplicationScope(final Configuration conf, final TableD
     });
   }
 
+  private static void checkDateTieredCompactionForTimeRangeDataTiering(final Configuration conf,
+    final TableDescriptor td) throws IOException {
+    // Table level configurations
+    checkDateTieredCompactionForTimeRangeDataTiering(conf);
+    for (ColumnFamilyDescriptor cfd : td.getColumnFamilies()) {
+      // Column family level configurations
+      Configuration cfdConf =
+        new CompoundConfiguration().add(conf).addStringMap(cfd.getConfiguration());
+      checkDateTieredCompactionForTimeRangeDataTiering(cfdConf);
+    }
+  }
+
+  private static void checkDateTieredCompactionForTimeRangeDataTiering(final Configuration conf)
+    throws IOException {
+    final String DATE_TIERED_STORE_ENGINE =
```

Comment:
nit: make this a public constant somewhere in the date tiered compaction code, then import static here.